Frontend: upgrade Vue and switch to Ethers

mveytsman opened the pull request:
- Rebuilt the app on Nuxt
- Got ethers installed and working with static JsonRPC

# TODO
- [x] Working in metamask
- [x] Event subscriptions
- [x] Test event subs
- [x] Buying ads using ethers `signer`
- [x] Test buy ads
- [x] Network switching when not in metamask
- ~~Error handling for the Promise-based ethers api~~
- [x] Add button to unlock wallet @shazow 
- [x] Add param for buy widget spawn position @shazow 
- [x] Normalize `ad.owner` @shazow
- [ ] Nuxtify static version @mveytsman 
- [ ] `unknown account #0 (operation="getAddress", code=UNSUPPORTED\_OPERATION, version=providers/5.1.2)`
- [ ] Reinstate the CSP (before merging)
- [ ] Handle giant ethers error output better (maybe only render the human-readable message and show the rest on-hover or something?)

## Non-blocking
- [ ] Convert to ethers.utils.BN
- [ ] Kill GA tracking snippets

# Questions for @shazow 
- Ethers returns "homestead" as the mainnet network name instead of "main". Can this cause problems in the future if we hardcode this string?
- Did I understand your intentions here: https://github.com/thousandetherhomepage/ketherhomepage/compare/upgrade-vue?expand=1#diff-fa059326cb969bbb3af4440b534bbfb369e32c1b75513a56348dbe8b4b2522bfR33 (I know it's a very old TODO, just curious if you remember).
- Revisit the bugs we had with Web3's BigNumbers.

shazow commented:
1. I think homestead is fine: https://github.com/thousandetherhomepage/ketherhomepage/pull/50/files#diff-06c67406a4d044fe2c49714374e3dabd78d88ff60ec9a9aa11aecd483e7dcdabR15
2. Yea we should make a computed property that just returns renderable ads, so we don't generate DOM for invalid ads I guess? But also I don't think that's really possible. And we're gonna render empty ads as placeholder blocks later.
3. Let's switch to using ethers' bignumber implementation where possible. I migrated tests to that in #50.

shazow reviewed (changes requested):
Wanna remove the duplicate stuff from src in this PR so we can see the deltas?

shazow commented on components/App.vue:
```diff
@@ -0,0 +1,242 @@
+    },
+  },
+  main: {
+    //TODO this was called "main" in web3, but ethers calls mainnet "homestead". Will using the name homestead break during some furture eth upgrade?
```
Yea, change it to `homestead`, but maybe keep an alias with `main` just in case.

shazow reviewed (commented):
Some code in here looks useful too: https://github.com/bmiller59/vue-cli-plugin-ethers/blob/master/generator/templates/module/ethersConnect.js

mveytsman commented:
@shazow I thought I dropped my TODO in here since we last paired but I guess I have not:

Here's my todo from May 11th, updated for today

- [x] unsuported network in the bottom menu
- [x] error about provider
- [x] clicking buy triggers a vuex error. Instead it should prompt you to choose a wallet.
- [x] Migrate static footer links (FAQ, Press, )
- Metamask
	- [x] with metamask: handle ropsten and others
- [ ] Verify "show NSFW"
- [ ] Monitor events
- [x] Resize dragging things don't appear
- [ ] Make a purchase on Rinkeby in a non-symmmetric slot to make sure we didn't invert coordinates 
- [x] in metamask call it main not homestead

The main thing we need to do is get purchasing and events to work and actually test this. I think we will be able to deploy this after next pair.

shazow commented:
@mveytsman Pushed monitor events, but still need to be verified. Also fixed network switching in non-wallet.
